doc/api/assert.markdown

```diff
@@ -3,9 +3,10 @@
     Stability: 3 - Locked
 
 The `assert` module provides a simple set of assertion tests that can be used
-to test invariants and implement unit tests. While the `assert` module is
-generally intended for internal use by Node.js itself, it can be used by user
-code calling `require('assert')`.
+to test invariants and implement unit tests. It is recommended that users of
+Node.js choose a third party assertion library from [npm][] instead of relying
+on `assert`.  That being said, `assert` can be used by user code calling
```

Comment:
@sam-github I'm a bit confused by your comments and feel that I must be misunderstanding something, so perhaps you could help me understand by rephrasing?

> This statement should not be this strong. There is *no* commonly npm package in existence for asserting ***invariants*** in source. At least, not that I know of, can you point to one?
> ...
> OK, I see, searching for assert on npmjs.org gets you pages of results, none of which are useful to replace assert for invariants

Isn't [`assert`](https://www.npmjs.com/package/assert), for example, pretty much the same thing? That's the first result I get when I search `assert`.

FWIW, there's also [`invariant`](https://www.npmjs.com/package/invariant) which is dead simple (basically just like `assert()`).

> and none of which are a unit test framework (which is probably what the reader wanted).
> 
> The usage of assert we want to warn people away from is using it as a unit testing library, such as mocha, tap, and nunit,

Do people somehow confuse the `assert` module as being a testing framework or library? Or do you mean it shouldn't be used for assertions in userland tests period? I think it's fairly commonly used for assertions within a framework like Mocha.